core/config/project_settings.cpp

```diff
@@ -56,6 +58,85 @@ String ProjectSettings::get_resource_path() const {
 
 const String ProjectSettings::IMPORTED_FILES_PATH("res://.godot/imported");
 
+// Returns the features that a project must have when opened with this build of Godot.
+// This is used by the project manager to provide the initial_settings for config/features.
+const PackedStringArray ProjectSettings::get_required_features() {
+	PackedStringArray features = PackedStringArray();
+	features.append(VERSION_BRANCH);
+#ifdef REAL_T_IS_DOUBLE
+	features.append("Double-precision");
+#else
+	features.append("Single-precision");
+#endif
+	return features;
+}
+
+// Returns the features supported by this build of Godot. Includes all required features.
+const PackedStringArray ProjectSettings::_get_supported_features() const {
+	PackedStringArray features = get_required_features();
+#ifdef MODULE_MONO_ENABLED
+	features.append("C#");
+#endif
+	// Allow pinning to a specific patch number or build type by marking
+	// them as supported. They're only used if the user adds them manually.
+	features.append(VERSION_BRANCH "." _MKSTR(VERSION_PATCH));
+	features.append(VERSION_FULL_CONFIG);
+	features.append(VERSION_FULL_BUILD);
+	// For now, assume Vulkan is always supported.
+	// This should be removed if it's possible to build the editor without Vulkan.
+	features.append("Vulkan");
+	return features;
+}
+
+// Returns the features that this project needs but this build of Godot lacks.
+const PackedStringArray ProjectSettings::_get_unsupported_features(const PackedStringArray &p_project_features) const {
+	PackedStringArray unsupported_features = PackedStringArray();
+	PackedStringArray supported_features = _get_supported_features();
+
+	for (int i = 0; i < p_project_features.size(); i++) {
+		if (!supported_features.has(p_project_features[i])) {
+			unsupported_features.append(p_project_features[i]);
+		}
+	}
+	unsupported_features.sort();
+	return unsupported_features;
+}
+
+// Returns the features that this project needs but this build of Godot lacks, as a user-friendly string.
+// This is used by the project manager to display and warn about unsupported features.
+const String ProjectSettings::get_unsupported_features_str(const PackedStringArray &p_project_features) {
+	PackedStringArray unsupported_features = singleton->_get_unsupported_features(p_project_features);
+	String features_string = Variant(unsupported_features).operator String();
+	if (features_string.length() < 3) {
+		return "";
+	}
+	return features_string.trim_prefix("[").trim_suffix("]");
+}
```

Comment:
If this is going to be used only by Project Manager, I'd move this method to project manager and make `_get_unsupported_features()` public.